presentation/src/main/java/org/sagebionetworks/research/presentation/perform_task/PerformTaskViewModel.java

```diff
@@ -211,51 +222,72 @@ public TaskView getTaskView() {
         return taskView;
     }
 
-    public void goBack() {
-        LOGGER.debug("goBack called");
-        Step currentStep = currentStepLiveData.getValue();
-        TaskResult taskResult = taskResultLiveData.getValue();
-        checkState(currentStep != null);
-        checkState(taskResult != null);
+    /**
+     * Sets the value of taskProgress, currentStep, and stepView live datas to match switching the step to the
+     * given next step.
+     * @param nextStep The step to use as the new current step.
+     * @param taskResult The task result before this switch occured.
+     */
+    protected void updateCurrentStep(@Nullable Step nextStep, @NonNull TaskResult taskResult) {
+        nextStep = resolveSection(nextStep);
+        if (nextStep == null) {
+            this.currentStepLiveData.setValue(null);
+            this.stepViewLiveData.setValue(null);
+            this.taskProgressLiveData.setValue(null);
+        } else {
+            TaskProgress nextProgress = this.stepNavigator.getProgress(nextStep, taskResult);
+            this.taskProgressLiveData.setValue(nextProgress);
+            LOGGER.debug("Setting step: {}", nextStep);
+            this.currentStepLiveData.setValue(nextStep);
+            StepView stepView = this.stepViewMapping.get(nextStep);
```

Comment:
There is a tradeoff here. If we do the transformation the app will launch tasks faster because it won't have to convert every step to a step view before it is needed, however it will make it so it is possible to have a malformed step that can't be converted to a step view (ie referencing a non-existent image) and this will only be caught when we try to run the step. Personally the app is launching really slow so I favor the transformation, what are your thoughts.